core/templates/services/playthrough.service.ts

```diff
@@ -171,16 +178,19 @@ class MultipleIncorrectAnswersTracker {
   providedIn: 'root'
 })
 export class PlaythroughService {
-  private explorationId: string = null;
-  private explorationVersion: number = null;
-  private learnerIsInSamplePopulation: boolean = null;
-
-  private eqTracker: EarlyQuitTracker = null;
-  private cstTracker: CyclicStateTransitionsTracker = null;
-  private misTracker: MultipleIncorrectAnswersTracker = null;
-  private recordedLearnerActions: LearnerAction[] = null;
-  private playthroughStopwatch: Stopwatch = null;
-  private playthroughDurationInSecs: number = null;
+  // The properties below are set to null when the service is initialized.
```

Comment:
@JeeveshGarg, to clarify -- I think the idea is to still keep these non-null in the static checks (and to refactor the code if necessary to ensure that that is the case). Just leave the runtime checks as they were, as an extra safeguard.

Reply:
1. We can do, but is it make sense, because we generally use `!` when we are sure we intialize them later, but here it may not a case. we may submit our recording action before it actually starts.

2. Also why i use null, because it indicates that recording and scrutinizing playthroughs is not enabled yet.

I think above both are correct but 2nd point make more sense.

Reply:
For (1) please check with @vojtechjelinek.

For (2) I don't think that's an adequate reason. If this component loads in the exploration editor then we should populate it with the proper values on initialization; this is orthogonal to whether it's turned on or not. For example it doesn't really make sense for an exploration ID to not exist in any context where this service is loaded.

Reply:
I think the non-null checks are not needed.

Reply:
@JeeveshGarg Hey, we've discussed this with Sean and I think we should go with the second approach. 

You should add the non-null checks and then also the `if` checks in the code. The reason for that is that we cannot be sure that in some non-strict files someone sends or sets `null` to this file and then we would face a runtime error risk.

You should also add TODO comments before the `if` check that will say that these `if` checks can be removed after we have strictly typed the whole codebase.